### PR TITLE
Use AcceptMessage for interop.java

### PR DESCRIPTION
**Pull request by grimmerm**
and convert incoming foreign values to Java values.

### CODE REVIEW

File: truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java [hunk 1]
```diff
@@ -891,8 +891,9 @@ private static void appendImports(Writer w) throws IOException {
             w.append("import com.oracle.truffle.api.nodes.RootNode;").append("\n");
         }
 
-        private void appendSingelton(Writer w) throws IOException {
+        private void appendSingeltonAndGetter(Writer w) throws IOException {
```
Comment (woess) [inline truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java]: **singleton**

File: truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/JavaObject.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * questions.
+ */
+package com.oracle.truffle.api.interop.java;
+
+import com.oracle.truffle.api.dsl.UnsupportedSpecializationException;
+import com.oracle.truffle.api.interop.ForeignAccess;
+import com.oracle.truffle.api.interop.TruffleObject;
+
+final class JavaObject implements TruffleObject {
+    @SuppressWarnings("unused") private static final UnsupportedSpecializationException QUIET_CANNONICALIZEPROJECTS = null;
```
Comment (woess) [inline truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/JavaObject.java]: ew! can we not have this please?

File: truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/JavaFunctionObject.java [hunk 3]
```diff
@@ -24,26 +24,27 @@
...
-    public Object execute(VirtualFrame frame) {
-        JavaInterop.JavaFunctionObject f = (JavaInterop.JavaFunctionObject) readField.execute(frame);
-        List<Object> args = ForeignAccess.getArguments(frame);
-        return JavaFunctionNode.execute(f, args.subList(1, args.size()).toArray());
+    public ForeignAccess getForeignAccess() {
+        return JavaFunctionObjectForeign.createAccess();
     }
 
-}
+} // end of JavaFunctionObject
```
Comment (woess) [inline truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/JavaFunctionObject.java]: I'd rather not have these comments

File: mx.truffle/suite.py [hunk 4]
```diff
@@ -338,6 +339,15 @@
         with most recent requirements of the Truffle infrastructure and tooling.""",
     },
 
+    "INTERNAL_PROCESSOR" : {
```
Comment (woess) [inline mx.truffle/suite.py]: I'd suggest `TRUFFLE_DSL_PROCESSOR_INTERNAL` or similar.

File: truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/JavaFunctionObject.java (hunk 3 above)
Comment (jtulach) [inline truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/JavaFunctionObject.java]: I prefer // end of class comments, as they help to understand what the closing brace is closing.

Comment (woess) [inline truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/JavaFunctionObject.java]: then you rename the class and the comment doesn't match anymore...

**Comment by jtulach** [comment]
When doing second commit to address Eclipse complains, I believe it would help to use "git commit --amend" to avoid clutter in the git history.

**Comment by jtulach** [comment]
The changes are backward compatible, but they introduce new API elements. Please make them package private:
```
Checking all signature changes against /home/devel/NetBeansProjects/truffle/truffle/com.oracle.truffle.api.interop/snapshot.sigtest

Added Methods
-------------

com.oracle.truffle.api.interop.UnsupportedTypeException:    method public final static java.lang.RuntimeException com.oracle.truffle.api.interop.UnsupportedTypeException.raise(java.lang.Exception,java.lang.Object[])

Checking all signature changes against /home/devel/NetBeansProjects/truffle/truffle/com.oracle.truffle.api.interop.java/snapshot.sigtest


Added Classes
-------------

com.oracle.truffle.api.interop.java.ToJavaNode
com.oracle.truffle.api.interop.java.ToJavaType
```
Or is this what you wanted to add?

**Comment by grimmerm** [comment]
@jtulach , yes, I wanted to have this second constructor.
`UnsupportedTypeError` are often caused by `UnsupportedSpecializationException`s.
However, if you are against changing API at all, I'm also fine with removing the second constructor.

**Comment by jtulach** [comment]
If the constructor change is intended then it is OK to change API.

**Pull request merged**